adapters/amqp-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -1097,9 +1095,11 @@ protected void onCommandReceived(
 
         if (sender.sendQueueFull()) {
             log.debug("cannot send command to device: no credit available [{}]", command);
-            commandContext.release(new ServerErrorException(HttpURLConnection.HTTP_UNAVAILABLE,
-                    "no credit available for sending command to device"));
+            final var exception = new ServerErrorException(HttpURLConnection.HTTP_UNAVAILABLE,
+                    "no credit available for sending command to device");
+            commandContext.release(exception);
```

Comment:
if I am not mistaken then the context will be released by the code invoking this method already ...
FMPOV either this method is responsible for accepting/rejecting the message or the calling code is, but not a mixture of both. In the latter case, we should probably pass in the command only but not the CommandContext ...
It looks like this decision needs to be made for the other protocol adapters as well.

Reply:
I was referring to line 1027 of this (VertxBasedAmqpProtocolAdapter) class. my understanding is that the `otherwise()` method will kick in here when the *onCommandReceived()* method  returns a failed future (as in line 1104), or am I mistaken?

Reply:
Oh, yes, that change was wrong there. I've pushed a fix.